### PR TITLE
Fixed #6594 - Calendar doesn't update visually when NOT using "Shared Calendar Separate"

### LINKED ISSUE #6594: Calendar doesn't update when NOT using "Shared Calendar Separate"
When using the calendar module with the `shared calendar separate` option **disabled** (=show all the calendars mixed into one), if you try to modify an entry assigned to someone else, the new data gets received by SuiteCRM but the on-screen page doesn't get updated.
This can be reproduced even on the current SuiteCRM Demo instance:

1. log into demo.suiteondemand.com (will / will)
2. select Calendar module
3. press "shared month"
4. press "settings" and disable "shared calendar separate" 
5. press "user list" and select everyone
6. press on an entry by someone else (purple color)
7. change the subject and save
8. notice that on the calendar the subject has not been updated

I think one problem is that inside `modules/Calendar/Cal.js`, function `CAL.dialog_save` there is code like:

`$("#calendar" + res.user_id).fullCalendar("removeEvents", res['record']);`
`$('#calendar' + res.user_id).fullCalendar('renderEvent', newEvent);`

When using the calendar in "unified" mode the only `#calendar[userid]` present in the DOM is the one of the current user, so the calls to removeEvents and renderEvent fail.

SuiteCRM 7.10.11, Firefox 63.0.3

### PR DESCRIPTION
**Pull request by Jorilx**
This patch checks to see if the user is NOT using "shared calendar separate" (by simply checking how many `#calendar[id].fc`s are present in the DOM), and if that's the case it directs updates to the only available (shared/mixed) calendar.

Please see the related issue for the steps to reproduce.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

File: jssource/src_files/modules/Calendar/Cal.js [hunk 1]
```diff
@@ -651,7 +651,18 @@ CAL.dialog_save = function () {
         CAL.update_vcal();
 
         var newEvent = new Object();
-        $("#calendar" + res.user_id).fullCalendar("removeEvents", res['record']);
+        // If there's more than one calendar ("shared calendar separate" is active) use the one related to the event user
+        var thisCal = $('div[id^="calendar"].fc');
```
Comment (gymad) [inline jssource/src_files/modules/Calendar/Cal.js]: css selector changed, please test original behavior before assess.

**Comment by jack7anderson7** [comment]
Hi @Jorilx,
When testing I found that after refreshing the page then the subject changes to what it was set as, is this now the case for you?

**Comment by Jorilx** [comment]
Do you mean that if you change the subject from "A" to "B" you **do** see "B", but when you refresh the page you get "A" again?

**Comment by jack7anderson7** [comment]
No I mean that once the subject is changed it doesn't change from "A" to "B" unless you refresh the page.

**Comment by Jorilx** [comment]
I see, which setting are you testing right now? "Shared calendars separate" on or off?

**Comment by jack7anderson7** [comment]
Off.

**Comment by Jorilx** [comment]
With more than one user selected from "user list"? Ok, I'll re-check the patch right away

**Comment by jack7anderson7** [comment]
Okay, hopefully we get the same result and yes I followed the instructions from the original issue.

**Comment by jack7anderson7** [comment]
@Jorilx what was the result of the re-test?

**Comment by Jorilx** [comment]
I'm not seeing your result on my setup (=my events get updated immediately), so I'm building a brand new SuiteCRM instance to try to reproduce your issue

**Comment by jack7anderson7** [comment]
@Jorilx, are you still testing this on the demo as your original issue states?

**Comment by Jorilx** [comment]
No I'm setting up a local SuiteCRM instance with this `issue-6594` branch

**Comment by Jorilx** [comment]
Just a thought: after a "repair and rebuild" I had to force a complete reload from the browser and now it's working correctly... Could it be a caching issue?

**Comment by jack7anderson7** [comment]
It could be, I will look into it, so the issue is no longer an issue?

**Comment by Jorilx** [comment]
To be completely sure I've tested the patch from scratch again: the crucial part is running a "Repair JS files" to rebuild the minified `SuiteCRM/modules/Calendar/Cal.js`, I thought that "repair and rebuild" did that too... Anyway I confirm that the patch is working

**Comment by samus-aran** [comment]
Hey @Jorilx If you are required to do a RepairJS you will need to also commit the minified files to the PR so that when people fork this repo that the JS qorks out of the box without having to run any admin tools.
Would suggest to Rebuild Minified JS Files and then Rebuild JS Grouping Files.

**Comment by Jorilx** [comment]
Ok I'm on it

**Comment by samus-aran** [comment]
Need to be replicated for hotfix-7.8.x? @salesagility/suitecrm-core

**Comment by samus-aran** [comment]
Hey @Jorilx we tried to email you on a found email address but wasn't sure it was accurate. If you have a chance to either jump onto the [gitter channel](https://gitter.im/suitecrm/Lobby) or email ourselves directly community@suitecrm.com. Its in regards to the PR Party and you being an author of a merged PR :+1:

**Comment by samus-aran** [comment]
> Hey @Jorilx we tried to email you on a found email address but wasn't sure it was accurate. If you have a chance to either jump onto the [gitter channel](https://gitter.im/suitecrm/Lobby) or email ourselves directly [community@suitecrm.com](mailto:community@suitecrm.com). Its in regards to the PR Party and you being an author of a merged PR

Nevermind! We must of got the right one ;)

**Pull request merged**